use vectorised logo

twogee opened the pull request:
I'd like to remove downloaded.gif, error.gif, evicted.gif and searched.gif as well.
Maybe we could vote on colours for downloaded and evicted 😉 if green is too dark or fuchsia is too garish (I took it because it's one of "safe" colours).

twogee commented:
This fixes https://issues.apache.org/jira/browse/IVY-922

twogee commented:
This one seems to break a JIRA record for time of first response: 80000+ hours 😄 Beat that!

janmaterne commented:
ivy-lierre.svg: 
- looks good
- havent compared, I think asciidoc/images and doc/images are the same?
- why this name? why not simply logo.svg? (ok, I see: ivy-lierre.png ...) But there is logo.png with same content (only smaller borders)

ivy-report.css
- seems that here is only different formatting 

ivy-report.xsl
- formatting
- changed <img ..gif> to <svg>
- translated img/@art->svg//title and img/@title->svg//title
! couldnt see the content of the grafik (where is defined that searched/downloaded svg should be used?)

IvyLogo.java
- havent tested that generated code
- JavaDoc of the parameters of IvyLogo(int width, int height) are bad ("ditto")
- I am thinking about the code generation ...
  a) include the flamingo task into our build and remove the generated code (= keep the model (svg) and the generator)
  b) just keep that generated code
  c) reuse the svg (Java reads the svg-resource and interpretes the drawing)
  Because this is one-shot we could go with b) ...



Checked out the code and the report looks fine.
But a simple ivy:report didnt work, I changed build.xml:500 to use explicitly the new xsl
  <ivy:report todir="${ivy.report.dir}" xslfile="src/java/org/apache/ivy/plugins/report/ivy-report.xsl"/>
Maybe my local environment ...

janmaterne commented:
Violations of the license header:
* ivy-lierre.svg (missing, remove the generator comment)
* IvyLogo (missing)

twogee commented:
logo.png is for CredentialsUtil; replaced by IvyLogo

CSS contains changes for `#logo` and adds `svg.icon` (somehow Firefox cannot read dimensions from embedded SVGs)

XSL generates HTML5 that uses xlink to reuse SVGs

IvyLogo is a one-off conversion of SVG + some manual additions (cleaned casts and added max icon size). Using SVG directly requires Batik, which is an overkill.

Here's the build.xml I used for testing Ivy report

```
<project name="ivy" default="ivy-report" xmlns:ivy="antlib:org.apache.ivy.ant">
    <target name="init-ivy">
        <taskdef resource="org/apache/ivy/ant/antlib.xml"
                 uri="antlib:org.apache.ivy.ant" classpath="build/artifact/jars/ivy.jar"/>

        <ivy:configure override="true" file="src/example/bintray/ivysettings.xml"/>
    </target>

    <target name="resolve" depends="init-ivy">
        <ivy:resolve organisation="org.hibernate" module="hibernate" revision="3.2.6.ga" transitive="true"
                     inline="true" conf="default" resolveid="hibernate.resolve"/>
    </target>

    <target name="ivy-report" depends="resolve">
        <mkdir dir="build/images"/>
        <ivy:report todir="build/images"
                    conf="default" resolveid="hibernate.resolve"/>
    </target>
</project>
```

janmaterne commented:
Still open are:

IvyLogo.java

    JavaDoc of the parameters of IvyLogo(int width, int height) are bad ("ditto")

test the report 
    But a simple ivy:report didnt work, I changed build.xml:500 to use explicitly the new xsl
    <ivy:report todir="${ivy.report.dir}" xslfile="src/java/org/apache/ivy/plugins/report/ivy-report.xsl"/>
    Maybe my local environment ...
    --> does the fresh built Ivy use the SVG graphics?

Violations of the license header:

    ivy-lierre.svg (header missing, remove the generator comment)
    IvyLogo (header missing)

twogee commented:
Thanks for recapitulating the issues; headers and javadoc corrected.

I tested the xsl using the buildfile provided above (without pointing out the xsl explicitly), so I tend to suspect you're picking up an old version of Ivy.
